app/(home)/page.tsx

```diff
@@ -229,20 +231,32 @@ function DashboardContent() {
 }
 
 export default function Dashboard() {
-  const { status, data } = useSession();
+  const { status } = useSession();
+  const [isAuthenticated, setIsAuthenticated] = useState(
+    status === 'authenticated'
+  );
+
+  useEffect(() => {
+    if (status === 'authenticated') {
+      setIsAuthenticated(true);
+    } else if (status === 'unauthenticated') {
+      setIsAuthenticated(false);
+    }
+  }, [status]);
```

Comment:
I'm not quite following the purpose of these changes, why are we setting the props to state rather than just determining the auth state directly from props?

```suggestion
  const isAuthenticated = status === 'authenticated';
```

Reply:
Right, I suppose line 253 `!isAuthenticated && status === 'loading'` relies on the previous render's `status`, I'm still not totally sure why though, some comments would be useful